### DISCUSSION TITLE
Expand TreeView so SelectedItem is visible

### DISCUSSION THREAD

**Discussion by dariostrm** [AvaloniaUI/Avalonia#16485] [Q&A]
Hello I have been trying to get the the items in a treeview to expand in order for the selected item in the tree view to be visible.
This is where I am at rn:
```c#
protected override void OnLoaded(RoutedEventArgs e)
    {
        base.OnLoaded(e);
        var selectedObject = ((SomeViewModel)DataContext!).SomeObject!;
        SomeTreeView.SelectedItem = selectedObject;
        List<IParentable> parentsFromTop = new();
        var parent = ObjectManager.Instance.GetParent(selectedObject);
        while (parent != null)
        {
            parentsFromTop.Insert(0, parent);
            parent = ObjectManager.Instance.GetParent(parent);
        }
        foreach (var parentFromTop in parentsFromTop)
        {
            var parentContainer = SomeTreeView.ContainerFromItem(parentFromTop);
            if (parentContainer is TreeViewItem treeViewItem)
            {
                treeViewItem.IsExpanded = true;
            }
        }
        SomeTreeView.ScrollIntoView(SomeTreeView.SelectedItem);
    }
```

This only expands the parent on the top level, and then the `ContainerFromItem` returns null when trying to get the parent on the next level. I think this is because there is no TreeViewItem created for it yet because it hasn't been made visible yet.
So I guess I somehow need to wait for everything to load after setting IsExpanded to true, but how do I do that? I didn't find any event...
Thank you in advance

**Comment by timunie** [comment]
I'd rather do it from ViewModel and bind IsExpanded in a style for TreeViewItem. There was an example anywhere in Q&A iirc but I can't find it rn.

**Reply by BhaaLseN** [reply]
You might be looking for #13475 (which I found during the same search as this Discussion). There's also  #11845 but the other one _should_ have the more simple solution for this (if it works for you).
